src/slices/posts.slice.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type Posts = {
+  loaded: boolean;
+  hasError: boolean;
+  items: Post[];
+};
+
+const initialState: Posts = {
+  loaded: true,
```

Comment:
The initial state for 'loaded' should be false, as the posts are not loaded initially. Setting it to true might cause incorrect UI behavior indicating that posts are loaded when they are not.